source/Calamari.ConsolidateCalamariPackages.Tests/IntegrationTests.cs

```diff
@@ -96,8 +100,77 @@ public void AndThenTheIndexShouldBe()
+            foreach (var reference in packageReferences.Where(pr => !pr.Name.Contains("Sashimi")))
+            {
+                var (flavour, platform) = ExtractFlavourAndPlatform(reference);
+                var outputFilename = Path.Combine(temp, $"{flavour}_{platform}_output.zip");
+                using (var outputStream = File.OpenWrite(outputFilename))
+                {
+                    using (var dest = new ZipWriter(outputStream, new ZipWriterOptions(SharpCompress.Common.CompressionType.Deflate) { DeflateCompressionLevel = CompressionLevel.BestSpeed, LeaveStreamOpen = false }))
+                    {
+                        foreach (var entry in consolidatedPackage.ExtractCalamariPackage(flavour, platform))
+                        {
+                            dest.Write(entry.destinationEntry, entry.sourceStream);
+                        }
+                    }
+                }
+
+                ZipFilesShouldBeIdentical(reference.PackagePath, outputFilename);
+            }
+        }
+
+        void ZipFilesShouldBeIdentical(string inputFilename, string regeneratedZipFilename)
+        {
+            using (var inputZip = ZipFile.OpenRead(inputFilename))
+            {
+                var sourceEntries = inputZip.Entries.Where(e =>
+                                                               !e.FullName.StartsWith("_rels") && !e.FullName.StartsWith("package") && !e.FullName.Equals("[Content_Types].xml")
+                                                          )
+                                            .ToList();
+                using (var regenZip = ZipFile.OpenRead(regeneratedZipFilename))
+                {
+                    //NOTE: some files appear multiple times in the regenerated zip file
+                    var regenNames = regenZip.Entries.Select(e => e.FullName).Distinct().ToList();
+                    var sourceNames = sourceEntries.Select(e => e.FullName).ToList();
+                    var missingNames = sourceNames.Where(s => !regenNames.Contains(s)).ToList();
+                    var addedNames = regenNames.Where(s => !sourceNames.Contains(s)).ToList();
+                    sourceNames.Should().BeEquivalentTo(regenNames);
+                }
+            }
+        }
+
+        static (string flavour, string platform) ExtractFlavourAndPlatform(BuildPackageReference packReference)
+        {
+            if (IsNetfx(packReference.Name))
+            {
+                return (packReference.Name, "netfx");
+            }
+
+            var packageName = packReference.Name;
+            var flavour = packageName.Split(".")[0];
+            var platform = packageName.Substring(flavour.Length).Trim('.');
+
+            return (flavour, platform);
+        }
+
+        static bool IsNetfx(string packageId)
+        {
+            return packageId.Equals("Calamari") || packageId.Equals("Calamari.Cloud");
+        }
+
+        private static string SanitiseFilenamesInIndex(string s)
```

Comment:
Driveby: why do filenames need to be sanitized?

Reply:
this can be done better - basically - whenever we see a hash - we want to replace it with a fixed tag.